inc/notificationtarget.class.php

```diff
@@ -167,15 +167,24 @@ function getFromDBForTarget($notifications_id, $type, $ID) {
     * @param boolean $notify_me notify me on my action ?
     *                           ($infos contains users_id to check if the target is me)
     *                           (false by default)
+    * @param int|null $emitter  if this action is executed by the cron, we can
+    *                           supply the id of the user who triggered the event
+    *                           so it can be used instead of getLoginUserID
     *
     * @return boolean
    **/
-   function validateSendTo($event, array $infos, $notify_me = false) {
+   function validateSendTo($event, array $infos, $notify_me = false, $emitter = null) {
+      $users_id = Session::getLoginUserID(false);
+
+      // Override session ID with emitter ID if supplied
+      if (!is_null($emitter)) {
+         $users_id = $emitter;
+      }
```

Comment:
```suggestion
      // Override session ID with emitter ID if supplied
      $users_id = $emitter ?? Session::getLoginUserID(false);
```

Reply:
Still can't do that on 9.4 PR, need PHP 7 :(

Reply:
Oh, you're right, sorry